onnxruntime/core/graph/graph_utils.cc

```diff
@@ -565,6 +565,21 @@ const Node* FirstChildByType(const Node& node, const std::string& child_type) {
   return nullptr;
 }
 
+std::vector<const Node*> FindChildrenByType(const Node& node, const std::string& child_type) {
+  // find children and sort them by source argument index
+  std::vector<const Node*> children(node.OutputDefs().size(), nullptr);
+  for (auto it = node.OutputEdgesBegin(); it != node.OutputEdgesEnd(); it++) {
+    if (it->GetNode().OpType().compare(child_type) == 0) {
+      children[it->GetSrcArgIndex()] = &(it->GetNode());
+    }
```

Comment:
What if there are two edges from the same output idx to different children?

Reply:
Fixed in the FindChildrenByType. FindParentsByType doesn't need the change because there is at most one edge for each input arg